homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -658,6 +661,33 @@ def __init__(self, name, device, entry, unique_id, coordinator, description):
         self._attr_native_value = self._determine_native_value()
         self._attr_extra_state_attributes = self._extract_attributes(coordinator.data)
 
+    def _extract_value_from_attribute(self, state, attribute):
+        """Overridden to handle different time units for timedelta.
+
+        This ensures that the device reported native values are converted to the ones
+        as defined in the entity descriptions.
+
+        This needs to be done here as the parent class does not have access
+        to the entity description.
+        """
+        value = getattr(state, attribute)
+        if not isinstance(value, timedelta):
+            return super()._extract_value_from_attribute(state, attribute)
+
+        conversion_map = {
+            TIME_SECONDS: timedelta(seconds=1),
+            TIME_MINUTES: timedelta(minutes=1),
+            TIME_HOURS: timedelta(hours=1),
+            TIME_DAYS: timedelta(days=1),
+        }
+
+        native_unit = self.entity_description.native_unit_of_measurement
```

Comment:
isn't `self.entity_description.native_unit_of_measurement` optional? Should there be a None check?

Reply:
I guess this goes hand in hand with Martin's comment below maybe.

Reply:
I don't think that it should ever be none in our use case, even when the base class defines it as optional.

Reply:
But someone who is going to contribute in the future might not know, that even though it's optional in the base, that it's required for this integration 🤔.

Reply:
True. Maybe we should move the `entity_description` to be a part of the `XiaomiCoordinatedMiioEntity` and enforce that it's not None there?

Reply:
So what should we do about this now? I'd prefer letting it crash early if someone adds sensors that are not defining the unit, it is easier to locate in the logs than just a logger error/warning.

Reply:
If it only crashes at dev time, then I say let it crash. But then, let's crash hard. As in, remove the `if` below that lets the code return total_seconds() instead.

Reply:
Um, sorry, could you clarify what you mean? My understanding is that it will crash hard here, as if the native unit type is `None`, calling `__getitem__` on `conversion_map` will raise a KeyError.

Reply:
@rytilahti / @OGKevin did you resolve this discussion?

Reply:
Yes, this is fixed. I just updated the breaking change description to be more clear about what was changed, so I think this is ready to go?